src/v/raft/recovery_stm.cc

```diff
@@ -133,7 +133,7 @@ ss::future<> recovery_stm::do_recover(ss::io_priority_class iopc) {
     }
 
     // wait for another round
-    if (meta.value()->last_sent_offset >= lstats.dirty_offset) {
+    if (meta.value()->in_flight_offset >= lstats.dirty_offset) {
```

Comment:
Deserves a comment about how this is about avoiding double sends when synchronizing with replicate_entries_stm